### ISSUE TITLE
NFIE iRODS Zone File Browsing

### ISSUE THREAD

**Issue by mjstealey** [hydroshare/hydroshare#351]
For NFIE Release, iRODS zone file browsing and bagit workflows are priorities

- It is n​ot​expected to be a problem to implement file browsing of HydroShare, NFIE, or
generally any iRODS zone by NFIE summer camp June 1

**Comment by dtarb** [comment]
Can you please explain what the account/access control approach planned is, noting that #350 plans to retain a single proxy user.  File browsing has to be limited to content for which the user has access.  I suggest that iRODS have the following accounts
1.  Proxy account that holds only properly formed HydroShare resources. There is no access to this directly from any iRODS client other than Django.
2.  Individual user accounts on iRODS that can be used for saving of files.  Browsing needs to be into this iroDS user workspace.

I am worried that the use case and details for this have not been fully thought through (and not written down).  We need a plan for this, that is simple and will meet NFIE needs.  
2.

**Comment by ironfroggy** [comment]
Based on our discussions about timelines so far, I don't think option two there is viable. There are plenty of other things to squeeze in before NFIE and, as far as I am aware, no work has yet begun to sync users from Hydroshare to iRODS. That isn't something you want NFIE to depend on without more time to test and work out any bugs, either.

It was my understanding that there would be some iRODS volume into which NFIE could drop resource files and could then find them in iRODS. Because of the lack of individual users for iRODS, this would have be a public volume for Hydroshare. However, the individual users for NFIE participants to access iRODS could be manually setup, yes?

**Comment by alvacouch** [comment]
Ray and I met with NFIE modeling principals on Monday and they complained about filtering not being present. They considered this a show-stopper for use of HydroShare on June 1. But their filtering needs are quite minimal. They have a naming convention and want to be able to select "files" with the appropriate "names". Basically, they need an "ls"/"dir" command that they can tell people about and that people can use to select resources, by title pattern.

a) use of the proxy account is irrelevant to this. We don't need to change that scheme to give them what they want. (In fact, my advanced access control scheme does not require moving away from the proxy account, either; we should do that only if we want to invoke iRODS quotas).

b) we definitely will need individual accounts on iRODS for NFIE students to use to create files. These will be located in a NFIE zone that is HydroShare-browsable.

c) movement between zones will be via ingestion of NFIE-zone resources into the HydroShare zone, via a browser interface.

On Monday, we also discussed a fallback position to using native iRODS for much of NFIE. It will satisfy their needs. Ray was working on implementing this, last I checked. I gave NFIE principals the information they need to publish modeling results in iRODS. It seems -- at this point -- that the Azure workflow engine being developed by Microsoft is not going to work on time, due to the need to port Linux native models to Windows. . Thus the modeling people will dual-publish their results: in the iRODS NFIE zone and the Azure blob store. This way, the Azure workflow people can continue to work, but the NFIE students will have access to data from which they can build the summer institute products.


Alva L. Couch, Ph.D.
Senior Architect, CUAHSI Water Data Center
http://wdc.cuahsi.org
+1 339 221 5400 x 203
